pkg/insights/insightsclient/insightsclient.go

```diff
@@ -92,19 +89,19 @@ func IsHttpError(err error) bool {
 var ErrWaitingForVersion = fmt.Errorf("waiting for the cluster version to be loaded")
 
 // New creates a Client
-func New(client *http.Client, maxBytes int64, metricsName string, authorizer Authorizer, gatherKubeConfig *rest.Config) *Client {
+func New(client *http.Client, maxBytes int64, metricsName string, authorizer Authorizer, gatherConfigClient *configv1client.ConfigV1Client) *Client {
```

Comment:
I think you can  pass `configClient *configv1client.Clientset` as the last param to make it a little bit more simple.